src/popup/bookmarkModule.js

```diff
@@ -302,20 +302,34 @@ document.addEventListener('DOMContentLoaded', () => {
     removeBookmarkImages();
   });
 
-  elements.deleteAllBookmarksButton.addEventListener('click', () => {
+  elements.deleteSelectedBookmarksButton.addEventListener('click', () => {
     chrome.storage.sync.get({ bookmarks: [] }, (items) => {
       const bookmarksArray = items.bookmarks;
-      if (bookmarksArray.length === 0) {
-        showNotification('No Bookmarks Available', 'negative', 'snackbar-bookmarks');
+      const bookmarkDOMArray = Object.values(bookmarkDOM);
+      bookmarkDOMArray.forEach((checkbox) => {
+        if (checkbox.checked) {
+          bookmarksArray[bookmarksArray.indexOf(checkbox.id)] = null;
+        }
+      });
+      const temp = [];
```

Comment:
Use a better variable name, like:
```suggestion
      const updatedBookmarksArray = [];
```